pkg/exec/exec.go

```diff
@@ -0,0 +1,73 @@
+package exec
+
+import (
+	"fmt"
+	osexec "os/exec"
+	"strings"
+)
+
+var testMode bool
+var testPrograms map[string]string
+
+type TestResult struct {
+	command string
+	output  string
+	err     error
+}
+
+var testResults []TestResult
+
+func SetTestMode() {
+	testMode = true
+	testPrograms = make(map[string]string)
+}
+
+func AddTestProgram(path string) {
+	lastSlash := strings.LastIndex(path, "/")
+	basename := path[lastSlash+1:]
+	testPrograms[basename] = path
+}
+
+func AddTestResult(command string, output string, err error) {
+	testResults = append(testResults, TestResult{command, output, err})
+}
+
+func LookPath(program string) (string, error) {
+	if testMode {
+		path, ok := testPrograms[program]
+		if !ok {
+			return "", fmt.Errorf("Not found: %s", program)
+		}
+		return path, nil
+	}
+
+	return osexec.LookPath(program)
+}
+
+func Exec(cmd string, args ...string) (string, error) {
+	if testMode {
+		var command string
```

Comment:
Might be easy to read if we move L49-65 to TestExec() method and return TestExec(...) in case of testMode=true